### ISSUE TITLE
unique elastic index for each table

### ISSUE THREAD

**Issue by measwel** [masumsoft/express-cassandra#148]
Is it possible to have the driver create a separate, unique index for each table indexed with elastic search? I am asking, because currently I run into problems with fields which have the same name in different tables. Elastic search does not allow to map equally named fields in different mappings within the same index. The solution is either to rename the fields ( costly ) or to have separate indexes per table...

There is one more possibility; to store all entities in the same cassandra table, differentiate based on 'type' and store diverging collumn data into a stringified text field. This would be a weird solution and also costly in terms of refactoring.

**Comment by measwel** [comment]
My suggestion would be the following:

```
      elasticsearch: {
        apiVersion: '5.5',
        sniffOnStart: true,
        host: config.elastic_server,
        requestTimeout: 60000,
        indexPerTable: true <== create a separate index per cassandra table
      }
```

indexPerTable:
true: create indexes called keyspacename_tablename
false: create just one index called keyspacename ( default )

If indexPerTable is true, the driver should query the correct index automatically based on the cassandra table name of the associated database model.

This solution would eliminate the problem of clashing field names in different entities - which is quite a big problem if the data model is large.

In my case, I have 2 entities which both have a 'location' field. With just 1 index I would have to rename the field in one of the entities. The problem is - besides refactoring the application - that I currently query both entities and merge the results in 1 response. This allows me to use the 'location' property in the frontend regardless of the underlying object. In other words, I can treat all returned objects the same way when displaying the location field.

**Comment by measwel** [comment]
Dear Masum,

I have started to rework the driver to support multiple ES indexes. I will let you know when its ready or when I get stuck - whatever happens first :) If I can do it, I will give you the code.

**Comment by measwel** [comment]
Dear Masum,

Can you please provide me with an email address to which I can send the code of the modified driver? You can reach me at marek_karczewski@yahoo.com.au

**Comment by masumsoft** [comment]
Hi, sorry for a delayed response. I was really busy in the meantime. I’ll try to have a look at it. But it would be great if you sent a pull request instead. That would hep me figure out the diff easily on github.

**Comment by measwel** [comment]
I cloned the repo and created a branch in which I committed my changes. Please provide further instructions. How should I push and create the pull request?

**Comment by masumsoft** [comment]
git push origin your_branch_name

And then...

https://help.github.com/articles/creating-a-pull-request/

**Comment by measwel** [comment]
I dont have write access to the main repo. Must I fork first? If so, how?

**Comment by measwel** [comment]
SourceTree is giving me errors when pushing the branch, but the push seems successful; the branch is in the 'measwel' fork. I have created a pull request.

**Comment by measwel** [comment]
Dear Masum,

I have tried several times to upload the code to github. For some reason I am getting upload errors. Attached are the 3 modified files. I turned off auto-formatting for them, to minimize spacing issues. Can you please please look at them and if approved, integrate them into the main branch? You will see that the changes are actually quite minimal; multiple indexes are made per table instead of 1 index for all tables. I think this is a valuable functionality upgrade.

Thank you greatly for the wonderful driver!

[ESindexPerModel.zip](https://github.com/masumsoft/express-cassandra/files/1690128/ESindexPerModel.zip)

**Comment by masumsoft** [comment]
Integrated the changes with some modifications applied to it. The is available in v2.2.0

**Comment by measwel** [comment]
Great news! Thank you so much. Is the default index per table or 1 index for all models?
